lib/widgets/info_card.dart

```diff
@@ -6,104 +6,116 @@ import '../resources/theme.dart';
 class InfoCard<T> extends StatelessWidget {
   const InfoCard(
       {@required this.future,
-      @required this.builder,
-      this.onShowMore,
-      this.title,
-      this.showMoreButtonKey,
-      this.padding});
+        @required this.builder,
+        this.onShowMore,
+        this.showIfEmpty = false,
+        this.title,
+        this.showMoreButtonKey,
+        this.padding});
 
   final Function onShowMore;
   final Future<T> future;
   final Widget Function(T) builder;
   final String title;
   final ValueKey<String> showMoreButtonKey;
   final EdgeInsetsGeometry padding;
+  final bool showIfEmpty;
 
   @override
   Widget build(final BuildContext context) {
-    return Padding(
+
+    return FutureBuilder(
+        future: future,
+        builder: (final context, final snapshot) {
+          if (snapshot.connectionState == ConnectionState.done) {
+            if (snapshot.hasData) {
+              if ((snapshot.data is Map ||
+                  snapshot.data is Iterable) &&
+                  snapshot.data.isEmpty) {
+                return showIfEmpty ? cardWrapper(context, noneYet(context)) : const SizedBox();
+              }
+              return cardWrapper(context, builder(snapshot.data));
+            } else {
+              return showIfEmpty ? cardWrapper(context, noneYet(context)) : const SizedBox();
+            }
+          }
+          return const SizedBox(
+            height: 100,
+            child: Center(child: CircularProgressIndicator()),
+          );
+        });
+  }
+
+  Widget cardWrapper(
+      final BuildContext context,
+      final Widget cardContent
```

Comment:
Named parameters?

Reply:
Made them named parameters, add a @required annotation too, hope it is all ok